[DOC] Clarify that `*` can be used as a wildcard in ACL rules (#5397)
strimzi/strimzi-kafka-operator: documentation/api/io.strimzi.api.kafka.model.AclRule.adoc

* Literal names are taken exactly as they are specified in the `name` field.

> * Prefix names use the value from the `name` as a prefix, and will apply the rule to all resources with names starting with the value.

Reviewer: ```suggestion
* Prefix names use the `name` value as a prefix and then apply the rule to all resources with names starting with that value.
```

When `patternType` is set as `literal`, you can set the name to `*` to indicate that the rule applies to all resources.

> .Example ACL rule which allows the user to read messages from all topics

Reviewer: ```suggestion
.Example ACL rule that allows the user to read messages from all topics
```
